Assets/LeapMotionModules/UI/Interaction/InteractionBehaviour.cs

```diff
@@ -0,0 +1,692 @@
+    }
+
+    public override void ContactStay(Hand hand) {
+      EvaluateContactCloseness(hand);
+
+      OnContactStay(hand);
+    }
+
+    private void EvaluateContactCloseness(Hand hand) {
+      float handDistance = GetInteractionDistanceToPoint(hand.PalmPosition.ToVector3());
+      if (_contactingHandsCount == 0 || _closestContactingHand == null) {
+        _closestContactingHand = hand;
+        _closestContactingHandDistance = handDistance;
+      }
+      else {
+        if (handDistance < _closestContactingHandDistance) {
+          _closestContactingHand = hand;
+          _closestContactingHandDistance = handDistance;
+        }
+      }
+    }
+
+    public override void ContactEnd(Hand hand) {
+      _contactingHandsCount -= 1;
+      if (_contactingHandsCount == 0) {
+        _closestJustStoppedContactingHand = hand;
+      }
+
+      OnContactEnd(hand);
+    }
+
+    public void ObjectContactBegin(Hand hand) {
+      OnObjectContactBegin(hand);
+    }
+
+    public void ObjectContactStay(Hand hand) {
+      OnObjectContactStay(hand);
+    }
+
+    public void ObjectContactEnd(Hand hand) {
+      OnObjectContactEnd(hand);
+    }
+
+    #endregion
+
+    #region Grasping
+
+    private int _graspCount = 0;
+    private bool _moveObjectWhenGrasped__WasEnabledLastFrame;
+    private bool _wasKinematicBeforeGrab;
+
+    private IGraspedPoseController _graspedPositionController;
+    private IGraspedPoseController GraspedPoseController {
+      get {
+        if (_graspedPositionController == null) {
+          _graspedPositionController = new KabschGraspedPose(this);
+        }
+        return _graspedPositionController;
+      }
+      set {
+        _graspedPositionController = value;
+      }
+    }
+
+    private KinematicGraspedMovement    _kinematicHoldingMovement;
+    private NonKinematicGraspedMovement _nonKinematicHoldingMovement;
+
+    private void InitGrasping() {
+      _moveObjectWhenGrasped__WasEnabledLastFrame = moveObjectWhenGrasped;
+
+      _kinematicHoldingMovement = new KinematicGraspedMovement();
+      _nonKinematicHoldingMovement = new NonKinematicGraspedMovement();
+    }
+
+    private int _graspCountLastFrame = 0;
+    private List<InteractionHand> _graspingIntHandsCache = new List<InteractionHand>(4);
+    private List<Hand> _graspingHandsCache = new List<Hand>(4);
+    private List<Hand> _lastFrameGraspingHandsCache = new List<Hand>(4); // for OnMultiGraspEnd
+
+    private void FixedUpdateGrasping() {
+      if (!moveObjectWhenGrasped && _moveObjectWhenGrasped__WasEnabledLastFrame) {
+        GraspedPoseController.ClearHands();
+      }
+      _moveObjectWhenGrasped__WasEnabledLastFrame = moveObjectWhenGrasped;
+
+      _graspingHandsCache.Clear();
+      foreach (var hand in _graspingIntHandsCache.Query().Select(intHand => intHand.GetLeapHand())) {
+        _graspingHandsCache.Add(hand);
+      }
+
+      if (_graspCount > 0) {
+        if (_graspCountLastFrame == 0) {
+          OnObjectGraspBegin();
```

Comment:
Any reason this isn't dispatched within the OnGraspBegin callback?  Will this cause a frame delay?